inc/commonitilobject.class.php

```diff
@@ -2717,142 +2717,211 @@ static function processMassiveActionsForOneItemtype(MassiveAction $ma, CommonDBT
+         'table'              => $this->getTable(),
+         'field'              => 'date_mod',
+         'name'               => __('Last update'),
+         'datatype'           => 'datetime',
+         'massiveaction'      => false
+      ];
+
+      $newtab = [
+         'id'                 => '7',
+         'table'              => 'glpi_itilcategories',
+         'field'              => 'completename',
+         'name'               => __('Category'),
+         'datatype'           => 'dropdown'
+      ];
 
       if (!Session::isCron() // no filter for cron
           && isset($_SESSION['glpiactiveprofile']['interface'])
           && ($_SESSION['glpiactiveprofile']['interface'] == 'helpdesk')) {
-         $tab[7]['condition']         = "`is_helpdeskvisible`";
-      }
-
-      $tab[80]['table']               = 'glpi_entities';
-      $tab[80]['field']               = 'completename';
-      $tab[80]['name']                = __('Entity');
-      $tab[80]['massiveaction']       = false;
-      $tab[80]['datatype']            = 'dropdown';
-
-      $tab[45]['table']               = $this->getTable();
-      $tab[45]['field']               = 'actiontime';
-      $tab[45]['name']                = __('Total duration');
-      $tab[45]['datatype']            = 'timestamp';
-      $tab[45]['massiveaction']       = false;
-      $tab[45]['nosearch']            = true;
-
-      $tab[64]['table']               = 'glpi_users';
-      $tab[64]['field']               = 'name';
-      $tab[64]['linkfield']           = 'users_id_lastupdater';
-      $tab[64]['name']                = __('Last edit by');
-      $tab[64]['massiveaction']       = false;
-      $tab[64]['datatype']            = 'dropdown';
-      $tab[64]['right']               = 'all';
+         $newtab['condition']         = "`is_helpdeskvisible`";
+      }
+      $tab[] = $newtab;
+
+      $tab[] = [
+         'id'                 => '80',
+         'table'              => 'glpi_entities',
+         'field'              => 'completename',
+         'name'               => __('Entity'),
+         'massiveaction'      => false,
+         'datatype'           => 'dropdown'
+      ];
+
+      $tab[] = [
+         'id'                 => '45',
+         'table'              => $this->getTable(),
+         'field'              => 'actiontime',
+         'name'               => __('Total duration'),
+         'datatype'           => 'timestamp',
+         'massiveaction'      => false,
+         'nosearch'           => true
+      ];
+
+      $newtab = [
+         'id'                 => '64',
+         'table'              => 'glpi_users',
+         'field'              => 'name',
+         'linkfield'          => 'users_id_lastupdater',
+         'name'               => __('Last edit by'),
+         'massiveaction'      => false,
+         'datatype'           => 'dropdown',
+         'right'              => 'all'
+      ];
+
+      // Filter search fields for helpdesk
```

Comment:
was there: https://github.com/glpi-project/glpi/pull/1441/files/865f8f90436cac86a6d6b4b52921e311940578e8#diff-007edb12589cf3c8df3121cd7b19028aL2585

I do not see why this rule should only apply on tickets; that is why it has moved here. Was I wrong?